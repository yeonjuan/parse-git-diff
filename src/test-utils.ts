import Context from './context';

type Validator<Args extends any[] = any[]> = (...args: Args) => boolean;

function createBooleanTester<V extends Validator>(
  validator: V,
  result: boolean
) {
  return function validTester(cases: Parameters<V>[]): void {
    cases.forEach((c) => {
      expect(validator(...c)).toBe(result);
    });
  };
}

export function createValidTester<V extends Validator>(validator: V) {
  return createBooleanTester(validator, true);
}

export function createInvalidTester<V extends Validator>(validator: V) {
  return createBooleanTester(validator, false);
}

export function createContext(diff: string, initial: number = 1) {
  const context = new Context(diff);
  for (let i = 0; i < initial - 1; i++) {
    context.nextLine();
  }
  return context;
}
