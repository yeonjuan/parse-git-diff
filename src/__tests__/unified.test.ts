import { getFixture } from './test-utils';
import parseGitDiff from '../parse-git-diff';

describe('unified', () => {
  const fixture = getFixture('unified');

  it('parse `unified', () => {
    expect(parseGitDiff(fixture)).toMatchSnapshot();
  });
});
