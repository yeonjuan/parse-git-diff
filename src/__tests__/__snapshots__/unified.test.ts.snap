// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unified parse \`unified 1`] = `
{
  "files": [
    {
      "chunks": [
        {
          "changes": [
            {
              "content": "ddd",
              "lineAfter": 3,
              "type": "AddedLine",
            },
          ],
          "context": "bbb",
          "fromFileRange": {
            "lines": 0,
            "start": 2,
          },
          "toFileRange": {
            "lines": 1,
            "start": 3,
          },
          "type": "Chunk",
        },
      ],
      "path": "test.txt",
      "type": "ChangedFile",
    },
  ],
  "type": "GitDiff",
}
`;
